devito/function.py

```diff
@@ -966,26 +966,18 @@ def interpolate(self, expr, offset=0, u_t=None, p_t=None, cummulative=False):
         """
         expr = indexify(expr)
 
-        # Apply optional time symbol substitutions to expr
-        if u_t is not None:
-            time = self.grid.time_dim
-            t = self.grid.stepping_dim
-            expr = expr.subs(t, u_t).subs(time, u_t)
-
         variables = list(retrieve_indexed(expr))
         # List of indirection indices for all adjacent grid points
         subs, idx_subs = self._interpolation_indices(variables, offset)
         # Substitute coordinate base symbols into the coefficients
         rhs = sum([expr.subs(vsub) * b.subs(subs)
                    for b, vsub in zip(self.coefficients, idx_subs)])
-        # Apply optional time symbol substitutions to lhs of assignment
-        lhs = self if p_t is None else self.subs(self.indices[0], p_t)
 
-        rhs = rhs + lhs if cummulative is True else rhs
+        rhs = rhs + self.subs(self_subs) if cummulative is True else rhs
```

Comment:
if your LHS has lots of sympy objects in it, this is gonna be a bit slower than in master.

perhaps do 
```
lhs = self.subs(self_subs)
```
then use it where needed?

BTW, nitpicking: I'd call it ``lhs_subs``, as tbh "self" is the function, not the indexed object that you are substituting into

Reply:
>if your LHS has lots of sympy objects in it, this is gonna be a bit slower than in master.

LHS is always just `rec` or `src` so should be fine